File: 김현주/의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> clothes_map; // {옷의 종류, 갯수}
    vector<string> clothes_type; //옷의 종류
    
    for(vector<string> v : clothes) {
        //map에 key(종류) 존재하면
        if(clothes_map.find(v[1]) != clothes_map.end()) {
            clothes_map[v[1]]++;
        }
        else {
            clothes_map.insert(make_pair(v[1],1)); // 입는경우 + 이 종류의 옷을 아예 입지 않는경우
            clothes_type.push_back(v[1]);
        }
    }

    int count = clothes_type.size();
    for(int i = 0; i<count; i++) {
        answer *= clothes_map[clothes_type[i]]+1;
    }
    return answer-1;
}

File: 김현주/의상/의상.js
function solution(clothes) {
    let answer = 1;
    const map = new Map();

    clothes.map(cloth=>cloth[1]).forEach((type, index)=> { //각 종류별 의상 개수 구하기
        map.set(type,(map.get(type)||0)+1)
    })
    
    for(const value of map.values()) {
        answer *= value+1; //아예 안 입는 경우+1
    }
    
    return answer-1; //최소 한 개의 의상은 입어야 하므로 전부 안 입는 경우 제외
}